frontend/interfaces/software.ts

```diff
@@ -291,3 +291,13 @@ export const INSTALL_STATUS_ICONS: Record<SoftwareInstallStatus, IconNames> = {
   installed: "success-outline",
   failed: "error-outline",
 } as const;
+
+export type IHostSoftwareWithLastInstall = IHostSoftware & {
+  last_install: ISoftwareLastInstall;
```

Comment:
Looks like `IHostSoftware` already has a `last_install` field, is this required?

Reply:
On IHostSoftware it is nullable. So this plus the `hasLastInstall` type guard let us signal to TS that this field is present and not null thereby letting us skip some duplicative null-ish checks in subcomponents.